src/internet_identity/src/assets.rs

```diff
@@ -19,15 +21,43 @@ pub enum ContentType {
     SVG
 }
 
-pub fn for_each_asset(mut f: impl FnMut(&'static str, ContentEncoding, ContentType, &'static [u8], &[u8; 32])) {
+lazy_static! {
+    // The <script> tag that sets the canister ID and loads the 'index.js'
+    static ref INDEX_HTML_SETUP_JS: String = {
+        let canister_id = api::id();
+        format!(r#"var canisterId = '{canister_id}';let s = document.createElement('script');s.async = false;s.src = 'index.js';document.head.appendChild(s);"#)
+    };
 
-    let index_html = include_bytes!("../../../dist/index.html");
+    // The SRI sha256 hash of the script tag, used by the CSP policy.
+    // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
+    pub static ref INDEX_HTML_SETUP_JS_SRI_HASH: String = {
+        let hash = &sha2::Sha256::digest(INDEX_HTML_SETUP_JS.as_bytes());
+        let hash = base64::encode(hash);
+        format!("sha256-{hash}")
+    };
 
-    let assets: [ (&str, &[u8], ContentEncoding, ContentType); 8] = [
+    // The full content of the index.html, after the canister ID (and script tag) have been
+    // injected
+    static ref INDEX_HTML_STR: String = {
+        let index_html = include_str!("../../../dist/index.html");
+        let foo: String = INDEX_HTML_SETUP_JS.to_string();
+        let index_html = index_html.replace(
+            "<script id='setupJs'></script>",
```

Comment:
is it intended that the `script` is not set as `lazy` or meant for`type="module"`?

randomly noticed this, just in case it would be useful.

Reply:
Thanks! This was already here, so I left it as is. I'll create a ticket to look into it.

Reply:
https://dfinity.atlassian.net/browse/L2-233